src/components/TextInput/TextInput.tsx

```diff
@@ -144,10 +158,24 @@ export const TextInput = React.forwardRef<HTMLSpanElement, TextInputProps>(funct
             )}
             data-qa={qa}
         >
-            {props.multiline ? (
+            {isInnerLabelVisible && (
+                <span
+                    {...innerLabelProps}
+                    ref={innerLabelRef}
+                    className={b('label', innerLabelProps?.className)}
```

Comment:
When you cut text with '...' you need to add title attribute on element for better ux

Reply:
Agree, will do.
Also I will wrap ResizeObserver in something like throttle/debounce, to minimize impact on performance.